plugin-commons/src/main/java/org/sonarsource/sonarlint/core/plugin/commons/loading/SonarPluginRequirementsChecker.java

```diff
@@ -168,19 +168,28 @@ private static PluginRequirementsCheckResult checkUnsatisfiedPluginDependency(Pl
       // If we evaluate A before B, then A might be wrongly included
       // But I'm not aware of such case in real life.
       if (depInfo == null || depInfo.isSkipped()) {
-        LOG.debug("Plugin '{}' dependency on '{}' is unsatisfied. Skip loading it.", currentResult.getPlugin().getName(), required.getKey());
-        return new PluginRequirementsCheckResult(currentResult.getPlugin(), new SkipReason.UnsatisfiedDependency(required.getKey()));
+        return processUnsatisfiedDependency(currentResult.getPlugin(), required.getKey());
       }
     }
     var basePluginKey = plugin.getBasePlugin();
-    if (basePluginKey != null) {
-      var baseInfo = currentResultsByKey.get(basePluginKey);
-      if (baseInfo == null || baseInfo.isSkipped()) {
-        LOG.debug("Plugin '{}' dependency on '{}' is unsatisfied. Skip loading it.", currentResult.getPlugin().getName(), basePluginKey);
-        return new PluginRequirementsCheckResult(currentResult.getPlugin(), new SkipReason.UnsatisfiedDependency(basePluginKey));
-      }
+    if (basePluginKey != null && checkForPluginSkipped(currentResultsByKey.get(basePluginKey))) {
+      return processUnsatisfiedDependency(currentResult.getPlugin(), basePluginKey);
+    }
+    if (DataflowBugDetection.PLUGIN_ALLOW_LIST.contains(plugin.getKey()) && checkForPluginSkipped(currentResultsByKey.get(Language.PYTHON.getPluginKey()))) {
+      // Workaround for SLCORE-667
+      // dbd and dbdpythonfrontend require Python to be working
```

Comment:
This means that DBD plugin will be loaded any time Python plugin is present, even if the `enableDataflowBugDetection` flag is false. I don't think this is desired and could cause further problems in the future for IntelliJ.

I think we could add an additional condition here about the feature flag. WDYT?

Reply:
It shouldn't be a problem as long as Python is there. Nonetheless, I still added the additional check. I had to change some logic regarding the SkipReason. Can you have another look?